Tasks/ECRPullImage/TaskParameters.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { AWSConnectionParameters, buildConnectionParameters } from 'Common/awsConnectionParameters'
+import tl = require('vsts-task-lib/task')
+
+export const imageTagSource: string = 'imagetag'
+export const imageIdSource: string = 'imagedigest'
+
+export interface TaskParameters {
+    awsConnectionParameters: AWSConnectionParameters
+    imageSource: string
+    imageTag: string
+    imageDigest: string
+    repository: string
+}
+
+export function buildTaskParameters(): TaskParameters {
+    const parameters: TaskParameters = {
+        awsConnectionParameters: buildConnectionParameters(),
+        imageSource: tl.getInput('imageSource', true),
```

Comment:
Should these getInput calls be replaced with getInputRequired and getInputOrEmpty?

Reply:
Yep, I made this before I added those functions, they need to be added + the module needs to be gone over when strict is turned on. Will add.